pkg/catalog/install.go

```diff
@@ -0,0 +1,112 @@
+package catalog
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+
+	helmv2 "github.com/fluxcd/helm-controller/api/v2beta1"
+	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kjson "k8s.io/apimachinery/pkg/runtime/serializer/json"
+)
+
+// Writer takes a profile and writes it out into a medium.
+type Writer interface {
+	Output(prof *profilesv1.ProfileSubscription) error
+}
+
+// FileWriter is a Writer using a file as backing medium.
+type FileWriter struct {
+	Filename string
+}
+
+// Output writes the profile subscription yaml data into a given file.
+func (fw *FileWriter) Output(prof *profilesv1.ProfileSubscription) error {
+	e := kjson.NewSerializerWithOptions(kjson.DefaultMetaFactory, nil, nil, kjson.SerializerOptions{Yaml: true, Strict: true})
+	f, err := os.OpenFile(fw.Filename, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+	if err != nil {
+		return err
+	}
+	defer func(f *os.File) {
+		if err := f.Close(); err != nil {
+			fmt.Printf("Failed to close file %s\n", f.Name())
+		}
+	}(f)
+	if err := e.Encode(prof, f); err != nil {
+		return err
+	}
+	return nil
+}
+
+// StdoutWriter is a Writer using stdout as backing medium.
+type StdoutWriter struct{}
+
+// Output outputs the yaml generated content for a profile to stdout.
+func (s *StdoutWriter) Output(prof *profilesv1.ProfileSubscription) error {
+	e := kjson.NewSerializerWithOptions(kjson.DefaultMetaFactory, nil, nil, kjson.SerializerOptions{Yaml: true, Strict: true})
+	return e.Encode(prof, os.Stdout)
+}
```

Comment:
I don't know if we need the stdout writer. In the ticket description I wrote both examples as in "pick one we can live with for just now" 😀

Reply:
Ahhh. :D Super sorry, I thought this thing should be choosable:
> I should see a ProfileSubscription yaml generated (either in a file or stdout)

So I thought `or` here meant that it should be choosable. :) Anyways, the interface is useful in terms of testing.

Reply:
yeh the interface is cool, but i don't think we need the stdout just yet. we can add it back later if that changes